crates/wasmtime/src/runtime/gc/enabled/externref.rs

```diff
@@ -363,8 +440,12 @@ impl ExternRef {
         // so that we can get the store's GC store. But importantly we cannot
         // trigger a GC while we are working with `gc_ref` here.
         let gc_ref = self.inner.try_gc_ref(store)?.unchecked_copy();
-        let externref = gc_ref.as_externref_unchecked();
```

Comment:
Given the footgun of using `as_externref_unchecked` would it make sense to remove that method entirely?

Reply:
Sure, I can make a follow up PR to do that, don't think it will be hard at this point.